src/client/components/form-elements/ReviewCollapsiblePanel.jsx

```diff
@@ -1,62 +1,102 @@
 import React from 'react';
+import Scroll from 'react-scroll';
 import { connect } from 'react-redux';
 import _ from 'lodash';
 
-import ErrorableCheckbox from './ErrorableCheckbox';
-import { updateIncompleteStatus, updateVerifiedStatus, updateCompletedStatus } from '../../actions';
+import { ensureFieldsInitialized, updateIncompleteStatus, updateVerifiedStatus, updateCompletedStatus } from '../../actions';
+import * as validations from '../../utils/validations';
 
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
 
 /**
  * A component for the review section to validate information is correct.
  *
  * Required props
  */
 
-
 class ReviewCollapsiblePanel extends React.Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.handleSave = this.handleSave.bind(this);
+    this.handleNext = this.handleNext.bind(this);
     this.handleEdit = this.handleEdit.bind(this);
+    this.scrollToTop = this.scrollToTop.bind(this);
   }
 
   componentWillMount() {
     this.id = _.uniqueId();
   }
 
+  scrollToTop() {
+    scroller.scrollTo('topScrollReviewPanel', {
+      duration: 500,
+      delay: 2,
+      smooth: true,
+    });
+  }
+
   handleSave() {
     const currentPath = this.props.updatePath;
-    this.props.onUpdateSaveStatus(currentPath);
+    const formData = this.props.data;
+    const sectionFields = this.props.uiData.sections[currentPath].fields;
+
+    this.props.onFieldsInitialized(sectionFields);
+    if (validations.isValidSection(currentPath, formData)) {
+      this.props.onUpdateSaveStatus(currentPath);
+      this.scrollToTop();
+    }
+    this.scrollToTop();
+  }
+
+  handleNext() {
+    const currentPath = this.props.updatePath;
+    this.props.onUpdateVerifiedStatus(currentPath, true);
+    // TODO: find a better solution for this or a different implementation.
+    const delay = 100; // 0.1 second
+    setTimeout(() => this.scrollToTop(), delay);
   }
 
   handleEdit() {
     const currentPath = this.props.updatePath;
     this.props.onUpdateEditStatus(currentPath);
     this.props.onUpdateVerifiedStatus(currentPath, false);
+    this.scrollToTop();
   }
 
   render() {
     let panelAction;
-    let editButton;
-    let verifiedLabel;
+    let buttonGroup;
     let hiddenSection;
+    let scrollHelper;
     const currentPath = this.props.updatePath;
     const sectionsComplete = this.props.uiData.sections[currentPath].complete;
     const sectionsVerified = this.props.uiData.sections[currentPath].verified;
     const allSections = Object.keys(this.props.uiData.sections);
     const sectionIndexes = allSections.indexOf(currentPath);
     const prevPath = allSections[sectionIndexes - 1];
 
+    const ButtonEdit = (
+      <button
+          className="edit-btn primary-outline"
+          onClick={this.handleEdit}><i className="fa fa-pencil"></i> Edit</button>
```

Comment:
Let's remove the space before `Edit`, as I think the other Edit buttons do not have that.